src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/DeriveHearingCentreHandlerTest.java

```diff
@@ -351,17 +352,98 @@ void should_not_set_hearing_centre_if_already_exists() {
     }
 
     @Test
-    void should_set_default_hearing_centre_for_detained_appeals() {
+    void should_set_default_hearing_centre_for_detained_appeals_ada_or_aaa() {
         when(callback.getEvent()).thenReturn(SUBMIT_APPEAL);
         when(callback.getCaseDetails()).thenReturn(caseDetails);
         when(callback.getCaseDetails().getCaseData()).thenReturn(asylumCase);
         when(asylumCase.read(HEARING_CENTRE, HearingCentre.class)).thenReturn(Optional.empty());
         when(asylumCase.read(APPELLANT_IN_DETENTION, YesOrNo.class)).thenReturn(Optional.of(YesOrNo.YES));
+        when(asylumCase.read(DETENTION_FACILITY)).thenReturn(Optional.of(DetentionFacility.OTHER));
+        when(asylumCase.read(IS_ACCELERATED_DETAINED_APPEAL, YesOrNo.class)).thenReturn(Optional.of(YesOrNo.YES));
+        when(asylumCase.read(AGE_ASSESSMENT, YesOrNo.class)).thenReturn(Optional.of(YesOrNo.NO));
 
         deriveHearingCentreHandler.handle(PreSubmitCallbackStage.ABOUT_TO_SUBMIT, callback);
         verify(asylumCase, times(1)).write(HEARING_CENTRE, HearingCentre.HARMONDSWORTH);
     }
 
+    @Test
+    void should_derive_hearing_centre_from_detention_facility_name_from_prison() {
```

Comment:
You can use parameterized test to input couple of prison names and verify that HC is assigned correctly.

Reply:
Tests updated